zerver/views/message_fetch.py

```diff
@@ -263,21 +263,24 @@ def messages_in_narrow_backend(
         )
     )
 
-    builder = NarrowBuilder(user_profile, column("message_id", Integer), user_profile.realm)
-    if narrow is not None:
-        for term in narrow:
-            query = builder.add_term(query, term)
+    inner_msg_id_col = column("message_id", Integer)
+    query, is_search = add_narrow_conditions(
+        user_profile=user_profile,
+        inner_msg_id_col=inner_msg_id_col,
+        query=query,
+        narrow=narrow,
+        is_web_public_query=False,
+        realm=user_profile.realm,
+    )
 
     search_fields = {}
     with get_sqlalchemy_connection() as sa_conn:
         for row in sa_conn.execute(query).fetchall():
```

Comment:
Should we just use `.mappings()` instead of `.fetchall()`, so that we don't have to set the local variables conditionally, and just pass `row["rendered_content"]` and etc. directly to `get_search_fields`?

Reply:
That seems like potentially a good idea, yeah.

Reply:
updated the changes.